saleor/site/utils.py

```diff
@@ -0,0 +1,19 @@
+from django.conf import settings
+
+from .models import SiteSetting
+
+
+def get_site_settings(request):
+    if not hasattr(request, 'site_settings'):
+        site_settings_id = getattr(settings, 'SITE_SETTINGS_ID', None)
+        request.site_settings = get_site_settings_uncached(site_settings_id)
+    return request.site_settings
+
+
+def get_site_settings_uncached(site_id=None):
+    return SiteSetting.objects.get(pk=site_id)
+
+
+def get_setting_value(request, key):
```

Comment:
Replacing `Foo.bar` with `getattr(Foo, 'bar')` opens a realm of maintenance nightmare where no one is sure which fields are used and which are not. I know because I'm working on one such legacy project right now.

The whole point here is not to cache the instance. A single call to Postgres is not much slower than a call to Redis.

See: https://code.djangoproject.com/ticket/15894#comment:16